### PR TITLE
forceha: fix two issues when (1)stop vm from inside (2) force remove host

### PR DESCRIPTION
**Pull request by ustcweizhou**
### Description

This PR fixes two issues. see description of each commit

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -4498,7 +4498,7 @@ private void handlePowerOffReportWithNoPendingJobsOnVM(final VMInstanceVO vm) {
                         String.format("VM %s is at %s and we received a %s report while there is no pending jobs on it"
                                 , vm.getInstanceName(), vm.getState(), vm.getPowerState()));
             }
-            if(vm.isHaEnabled() && vm.getState() == State.Running
+            if((HighAvailabilityManager.ForceHA.value() || vm.isHaEnabled()) && vm.getState() == State.Running
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:4501]: @harikrishna-patnala yes,  it is a cluster-scope setting.
But cloudstack does not consider it before.
```
        _forceHA = ForceHA.value();
```

 same as other cluster-scope settings in https://github.com/apache/cloudstack/blob/master/engine/components-api/src/main/java/com/cloud/ha/HighAvailabilityManager.java.

I am wondering if we should fix all of them in a pr.

**Comment by rhtyd** [comment]
@blueorangutan test

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhtyd) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:4501]: @weizhouapache can you address the scope issue? Or change scope of config key to be global than cluster specific of it never supported cluster level setting.

**Comment by rhtyd** [comment]
ping @weizhouapache cc @ravening  @Pearl1594  @Spaceman1984 @shwstppr - should we consider this for 4.15.1, can you provide comments on testing/reviewing this?

**Comment by shwstppr** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (harikrishna-patnala) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:4501]: If we change it to global, then we may miss setting this back to cluster when we want to fix all other global settings in that Manager. This configuration parameter makes more sense to be scoped to cluster. May be as @weizhouapache mentioned, we need to fix all other settings in another PR. I'm fine with this for now.

**Review by harikrishna-patnala — APPROVED** [review]
Code LGTM

**Review by shwstppr — APPROVED** [review]
LGTM. Tested both 4.15.1-Snapshot and 4.14 branch with this PR.
Able to reproduce both issues on 4.15.
For this PR:

- VM got restarted when force.ha=true and VM was stopped from inside using console.
- VMs got stopped/domain removed on KVM host when host was forece-removed from CS

**Comment by weizhouapache** [comment]
@harikrishna-patnala @shwstppr cool, thanks for review and testing !

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
Failures not related to PR

**Pull request merged**